### PR TITLE
Update CLI interactive mode

### PR DESCRIPTION
**Pull request by monstruosoft**
CLI interactive mode now works with all supported image generation modes: txt2img, img2img, Image variations, EDSR upscale and SD upscale.

At first I attempted to write a basic _curses_-based interface but that resulted in messed up console output because of the rest of the code is not written with a _curses_ interface in mind, so in the end I decided to write a most simple interface. I've tried all possible modes and combinations (interactive and non-interactive, GUI and WebUI) and I think it's all working as expected.

Some CLI interactive modes, in particular SD upscale, might require some extra work to allow configuring custom settings.

If you have some coding style rules, let me know, so that you don't have to edit the changes just for coding style.

### CODE REVIEW

**Comment by rupeshs** [comment]
@monstruosoft Does this works on Windows (curses?)

**Comment by monstruosoft** [comment]
I'm not sure, don't have a Windows machine to try it, but in the end I didn't use curses, it's just plain console input and output routines.

File: src/frontend/cli_interactive.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+import curses
```
Comment (rupeshs) [inline src/frontend/cli_interactive.py:1]: Remove unused imports

File: src/frontend/cli_interactive.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        value = value_type(input(message))
+    except:
+        value = default_value;
+    return value
+
+
+
+def interactive_settings(config, context):
+    global _batch_count
+    c = config.lcm_diffusion_setting
```
Comment (rupeshs) [inline src/frontend/cli_interactive.py:58]: Avoid one-letter variable names. Use intention revealing variable names.

File: src/frontend/cli_interactive.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    elif option == 3:
+        interactive_variations(config, context)
+    elif option == 4:
+        interactive_edsr(config, context)
+    elif option == 5:
+        interactive_sdupscale(config, context)
+
+
+
+def user_value(value_type: type, message: str, default_value: Any) -> Any:
```
Comment (rupeshs) [inline src/frontend/cli_interactive.py:47]: Use trailing comma at the end of function arguments.
def user_value(value_type: type, 
                    message: str, 
                    default_value: Any,)
Use black code formatter

File: src/frontend/cli_interactive.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                settings=config,
+                device=DEVICE,
+            )
+        user_input = input("Write a prompt: ")
+
+
+
+def interactive_img2img(config, context):
+    global _batch_count
+    c = config.lcm_diffusion_setting
```
Comment (rupeshs) [inline src/frontend/cli_interactive.py:89]: Give a better name instead of c

File: src/frontend/cli_interactive.py (hunk 1 above)
Comment (rupeshs) [inline src/frontend/cli_interactive.py:1]: Remove unused imports

**Comment by rupeshs** [comment]
> I'm not sure, I don't have a Windows machine to try it, but in the end I didn't use curses, it's just plain console input and output routines.

@monstruosoft  I have added a few review comments.
For coding follow these style :
- Code formatted - black
- PEP 8 code style
- Use intention revealing names to arguments, functions, or classes
- Split larger functions/classes into smaller
- Follow the single responsibility principle
- SOLID principles
- Add a trailing comma 

You can also use ruff with IDE.

**Comment by monstruosoft** [comment]
> Give a better name instead of c

I used a single letter because I thought that would make the code shorter and easier to read; if I use a longer name I might as well just use the whole `config.lcm_diffusion_setting` instead, but OK, I'll review the code.

> Remove unused imports

Right, missed that one.

> Write code in a cross-platform way (Windows/Linux/Mac)

Isn't Python by itself cross-platform?.

Gonna make the necessary changes. Thanks.

**Comment by rupeshs** [comment]
> I've improved CLI interactive mode and used _black_ for code formatting. The code is not pretty but it allows customizing every possible useful setting I could think of, so that you can now use CLI interactive mode simply by running FastSD as:
> 
> `# python src/app.py -i`
> 
> I made a lot of test runs to make sure it works well, I hope I didn't miss anything.

Thanks, Could you please resolve all review comments?

**Comment by monstruosoft** [comment]
Done.

**Comment by rupeshs** [comment]
@monstruosoft Thank you for this wonderful PR.

**Pull request merged**